docs/guides/setting-up.md

```diff
@@ -0,0 +1,118 @@
+# Introduction
+
+This page explains how to perform a basic set up after installing Linux on a T2 Mac.
+
+## Do you need to do this?
+
+This guide is mainly relevent in the following cases :-
+
+1. If you have installed Linux using an official ISO, instead of a T2 ISO.
+2. The [Make modules load on early boot](#make-modules-load-on-early-boot) section is relevant for those who wish to encrypt their disk drives using LUKS or some other similar software.
+3. If some functionality related to T2 Macs is broken, then you can consider following this guide.
+
+In rest cases, you probably won't need to follow this guide.
+
+## Installing a kernel for T2 support
+
+Installing a kernel with support for T2 Macs is required in order to get the Keyboard, Trackpad, Touch Bar, Audio, Fan and Wi-Fi working.
+
+Many distro maintainers provide compiled kernels which can be installed on your Linux installation. Following are the links to the repos providing such kernels :-
+
+| Linux Distribution   | Kernel with T2 support |
+| -------------------- | ---------------------- |
+| Arch based distros   | https://github.com/Redecorating/linux-t2-arch |
+| Fedora               | https://github.com/mikeeq/mbp-fedora-kernel |
+| Gentoo               | https://github.com/t2linux/T2-Gentoo-Kernel |
+| Manjaro              | https://github.com/NoaHimesaka1873/manjaro-kernel-t2 |
+| NixOS                | TODO |
+| Ubuntu based distros | https://github.com/t2linux/T2-Ubuntu-Kernel |
+| Debian based distros | https://github.com/andersfugmann/T2-Debian-Kernel |
+
+If compiled kernels for your distro are not available, then you shall have to compile a kernel on your own. You can follow the [Kernel](https://wiki.t2linux.org/guides/kernel/) guide for help.
+
+## Add necessary kernel paramaters
+
+Using your bootloader, add the `intel_iommu=on iommu=pt pcie_ports=compat` kernel parameters. For example in GRUB :-
+
+  1. Edit `/etc/default/grub`.
+  2. On the line with `GRUB_CMDLINE_LINUX="quiet splash"`, add the following kernel parameters: `intel_iommu=on iommu=pt pcie_ports=compat`.
+  3. Run `sudo grub-mkconfig -o /boot/grub/grub.cfg` if you are on a non-debian based distro. If using Debian or Ubuntu based distro, run `sudo update-grub`.
+
+## Make modules load on boot
+
+Simply run the following :-
+
+```sh
+echo apple-bce | sudo tee /etc/modules-load.d/t2.conf
+```
+
+## Make modules load on early boot
+
+Having the `apple-bce` module loaded early allows the use of the keyboard for decrypting encrypted volumes (LUKS).
+It also is useful when boot doesn't work, and the keyboard is required for debugging.
+To do this, one must ensure the `apple-bce` module *as well as its dependent modules* are included in the initial ram disk.
+If your distro uses `initramfs-tools` (all debian-based distros), then `/etc/initramfs-tools/modules` stores a list of extra modules to be included and loaded at early boot time.
```

Comment:
for this section maybe we can do it as a list?

- on systems with initramfs-tools
    1. step 1
    2. step 2
- systems with mkinitcpio
    1. step 1
    2. step 2
- systems with other initramfs/initrd generation systems
    1. refer to the documentation and ensure the kernel module apple-bce is loaded early

or something like that.

also note that not all arch systems use mkinitcpio, some users may choose to use booster or other ones and will have to consult the respective documentation

Reply:
Done